### ISSUE TITLE
Support hot-reload for Wechaty events listeners

### ISSUE THREAD

**Issue by zixia** [Chatie/wechaty#820]
The API might add the following new features:

### 1. addListener

```ts
wechaty.on('message', './message-listener')
```

If the second parameter is a string instead of a function, then we can use `func = require('./message-listener')` as the listener function, monitor the file, and reload it if the file had been updated.

### 2. WechatyInstanceOptions

```
bot = Wechaty.instance({
    listenerDirectory: './listenerDirectory/'    // this is the directory which contains all listeners.
})
```

Wechaty will scan `./listenerDirectory/` for the files that the same name as the event, then `require` it and add it as the listener.

-------------

### See Also

1. https://github.com/fastest963/require-reload
1. http://blog.chatie.io/developer/2017/03/20/added-hot-reload-for-bots.html
1. https://blogs.msdn.microsoft.com/typescript/2017/06/12/announcing-typescript-2-4-rc/
1. https://www.youtube.com/watch?v=cGqfM6huJCs
1. [Getting the stack trace in a structured form](https://github.com/nodejs/node/issues/7749#issuecomment-232972234)
1. [Get v8 stack traces as an array of CallSite objects.](https://github.com/felixge/node-stack-trace)
1. [A node module for enabling a module to determine its caller.](https://github.com/totherik/caller)
1. [[Design Spec] ESNext import()](https://github.com/Microsoft/TypeScript/issues/14495)
1. [Hot module replacement for node.js](https://github.com/sidorares/hot-module-replacement)
1. [Hot reload tool for node.js](https://github.com/yyrdl/dload)

CC @Gcaufy

**Comment by Gcaufy** [comment]
Looks good for me.
There is another way to do it, may be more convenient to the developers
```
bot = Wechaty.instance({
    listeners: './listeners'    // this is the directory which contains all listeners.
});
bot.init();
```

May be you can support both ways.

**Comment by zixia** [comment]
@Gcaufy I just published a new NPM module as a Hot Module Reload tool named `hot-import` at here: https://github.com/zixia/hot-import

It supports TypeScript, please give it a try and let me know what you think. :)

**Comment by zixia** [comment]
Landed at v0.8.240.

**Comment by zixia** [comment]
Get to know more about the usage of hot-import listener in Wechaty: #923 #935
